web/src/settings_profile_fields.js

```diff
@@ -357,12 +343,25 @@ function set_up_external_account_field_edit_form($profile_field_form, url_patter
     }
 }
 
-function set_up_select_field_edit_form($profile_field_form, field_data) {
+function disable_submit_btn_if_no_property_changed($profile_field_form, field) {
+    const data = populate_data_for_request($profile_field_form, false, undefined, undefined, field);
+    let save_changes_button_disabled = false;
+    if (Object.keys(data).length === 0 || data.field_data === "{}") {
+        save_changes_button_disabled = true;
+    }
+    $("#edit-custom-profile-field-form-modal .dialog_submit_button").prop(
+        "disabled",
+        save_changes_button_disabled,
+    );
+}
+
+function set_up_select_field_edit_form($profile_field_form, field) {
     // Re-render field choices in edit form to load initial select data
     const $choice_list = $profile_field_form.find(".edit_profile_field_choices_container");
     $choice_list.off();
     $choice_list.empty();
 
+    const field_data = field.field_data ? JSON.parse(field.field_data) : {};
```

Comment:
Here also, I guess `field.field_data` would always be defined for select fields.

Reply:
Thanks, fixed at [diff link](https://github.com/zulip/zulip/compare/b0d08f767aad67ff0c691530e0b53881256a6c42..fd0f742681283a3ec7af684e59b34ce788c720fe).